doc_crosslinks_new/crosslink_redirect.html

```diff
@@ -0,0 +1,112 @@
+        
+            var uidServiceData = '${UID_SERVICE_MAPPING}';
+            
+            var uidToServiceName = JSON.parse(uidServiceData);            
+            
+            var baseUri = "LATEST/";
+            
+            function getQueryStringParameter(name, url) {
+                
+                var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+                name = name.replace(/[\[\]]/g, "\\$&");
+                
+                results = regex.exec(url);
+                if (!results) return null;
+                if (!results[2]) return '';
+                return decodeURIComponent(results[2].replace(/\+/g, " "));
+            }
+            
+            function escapeCharacters(toEscape) {
+                var escapedString = "";
+                
+                for(var i = 0; i < toEscape.length; ++i) {
+                    var character = toEscape.charAt(i);
+                    if(character != character.toLowerCase()) {
+                        escapedString += '_' + character.toLowerCase();
+                    }
+                    else if(character == ':') {
+                        escapadString += "_1";
+                    }
+                    else {
+                        escapedString += character;
+                    }
+                }
+                
+                return escapedString;
+            }
+            
+            function transformRequestToDoxygenLink(url, typeName, operation) {                
+                var uid = getQueryStringParameter("uid", url);                
+                
+                var serviceName = uidToServiceName[uid];
+                
+                if(serviceName == null) {
+                    return baseUri;
+                }
+                
+                var dirtyName = "class_aws_1_1_";
+                dirtyName += serviceName.charAt(0).toLowerCase() + serviceName.substring(1);
+                
+                if(typeName != null) {                    
+                    dirtyName += "_1_1_model_1_1_";
+                    var firstCharLowered = typeName.charAt(0).toLowerCase() + typeName.substring(1);
+                    var operationCanonicalizedStr = firstCharLowered.replace("Input", "Request").replace("Output", "Result");
+                    
+                    dirtyName += operationCanonicalizedStr;
+                }
+                else if(operation != null) {
+                    dirtyName += "_1_1_";
+                    var firstCharLowered = serviceName.charAt(0).toLowerCase() + serviceName.substring(1);
+                    dirtyName += firstCharLowered;
+                    dirtyName += "_client";
+                    //once we figure out the operation link, add the #stuff here:
+                    //dirtyName += "#operationName";
+                }
+                
+                dirtyName += ".html";
+                return baseUri + serviceName + "/html/" + escapeCharacters(dirtyName);
```

Comment:
this is the only line that changes with the old creating doc crosslinks, otherwise it is a direct copy, i created a copy to preserve backwards compatibility for now so that a rollback wouldn't be difficult.